### ISSUE TITLE
reload does not keep the seed value

### ISSUE THREAD

**Issue by sriramab** [gama-platform/gama#1614]
Hi,

When you fix a seed value in experiment, and also have a parameter value: First load of model is not a problem. But when you change the parameter value and hit reload, the seed value gets changed. 

I would prefer the seed value to remain unchanged, because some agents are initialised based on seed value for replication.

regards,
Srirama

**Comment by AlexisDrogoul** [comment]
Hi Srirama,

We are (slowly) updating the documentation and there is something new in GAMA 1.7, which is that you can define a seed for the experiment and a seed for each simulation. 

Regarding the parameters view, you are also normally able to fix the seed to a given value.

Does it answer your question ? 

Alexis

**Comment by sriramab** [comment]
Hi 

I understand what you mean. But my question is simple. I define an experiment with a seed. I load the model. The seed value is correct in the parameter view. In parameter view, I change a variable and reload the model from parameter view (with the anticlockwise circular arrow). Now the seed is changed. I don't want the seed value to change. 

Do you confirm that even in this case, only one experiment, it will change? 

I do not understand what you mean by seed for experiment and seed for simulation. 
I have my seed in experiment section as a float. I guess this is experiment seed. 
What is simulation seed?

Please could you clarify?
regards,
Srirama

**Comment by AlexisDrogoul** [comment]
In GAMA 1.7, since an experiment can now manage several simulations in parallel, it has its own seed (for generating random values), and each simulation can be provided with its own seed (to generate random values by itself — in that sense, you are able to run simulations with different values of seed and compare them side by side. 

So now, if you want to fix a seed for your simulation, you have to define it in the simulation (i.e. in global, not in the experiment).

**Comment by sriramab** [comment]
Oke, then it is still a problem. If I define it in the global section. The random seed is not kept. The values of my variables change every run even with a global definition of seed.
please see video
[seed.zip](https://github.com/gama-platform/gama/files/162833/seed.zip)

**Comment by AlexisDrogoul** [comment]
And the `init: … ` you are displaying are the result of something you compute in the simulation ?

**Comment by sriramab** [comment]
Yes, it is distribution of 306 elements as you can see in the pie chart. I would like that the init:- or what I use to address the initial configuration of 306 elements. And with a seed value in global these change, while a seed value in experiment keeps the same configuration.

**Comment by sriramab** [comment]
Please see here, I have placed seed in experiment and the value keeps my init (initial configuration) always the same, as I expect. Also notice the seed displayed in parameter windows is same as experiment seed and not when defined in global seed.
[seedExperiment.zip](https://github.com/gama-platform/gama/files/162875/seedExperiment.zip)

**Comment by AlexisDrogoul** [comment]
Yep. You're right. I have identified the problem: simulations are passed on, by default, when they are created, the current seed of the experiment. And this value prevents them to define their own seed (actually, the only way to force a new seed right now is to write something like: 
```
create simulation with: [seed::100];
```
)
I will issue a patch ASAP.

Alexis

**Comment by sriramab** [comment]
And that statement goes in global init along with other create species statements?
This is confusing :(

**Comment by AlexisDrogoul** [comment]
No, in `experiment` (in a `reflex` or a `init`, for instance). You can have a look at the model called `Ants Foraging (4 simulations)`.

**Comment by sriramab** [comment]
Hmm, that model (dated 8 March 16) throws an error: java.util.concurrent.rejectedexecutionexception
However, you mean to say I should use seed as an argument of "with:" in create statement inside init of experiment?

**Comment by mazarsju** [comment]
Hi Srirama,

In the new documentation (still in developpement, it will be released with 1.7), there is a section about seed in multi-simulation : https://github.com/mazarsju/gama_doc_17/wiki/RunSeveralSimulations#random-seed

**Comment by AlexisDrogoul** [comment]
Hi,
 I have committed a fix for this behavior. Basically, now, if a seed/rng is defined in the global section, there is no attempt (from the experiment) to override it.

It should normally work, then.

Regarding the java.util.concurrent.rejectedexecutionexception, I am quite surprised as it should not happen at all (i.e. your trace indicates that you use an "AbortPolicy" when too many threads want to be executed at once, but this policy has been replaced one week ago (see #1601 )). Can you please open a new issue about this if you are certain that you are up to date regarding the repository ?

**Comment by sriramab** [comment]
Yes, my repository is older than a week. To update now, Now, I am getting a checkout conflict.
What should I do? I am on Mac Mavericks

Checkout conflict with files: 
msi.gama.application/gama1.7.Eclipse3_8_2.product
Checkout conflict with files: 
msi.gama.application/gama1.7.Eclipse3_8_2.product

**Comment by AlexisDrogoul** [comment]
OK. Did you change anything in your product on purpose ? 
If not, you can simply right-click on it and choose "Replace With > HEAD Revision". This will eliminate the conflict by replacing your file with its "pristine" state. 
If you have changed something meaningful, then you can copy and paste it (give the copy another name), and do the first procedure on the original.

**Comment by sriramab** [comment]
Thank you it seems to be working. I did not change anything, only added StartExplorer, CSVedit as recommended on wiki.  Are they not compatible with new GAMA GIT version?

**Comment by AlexisDrogoul** [comment]
OK. Next time you want to add plugins, remember you can do it also in the run configuration. Or you duplicate your product and create a new one, this works as well and it is safer. Regarding CSVedit, it shouldn't be needed anymore as we have an integrated CSV editor (mostly viewer, though). I'd be interested to see if the 2 work well together :)
